Fix SONiC compilation for bfnsde-9.0.0 and customize deployment on IMT devices.

YuriiHordynskyi opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- Why I did it**
To fix SONiC compilation for bfnsde-9.0.0 and customize deployment on IMT devices.
**- How I did it**
Modified image build, install, and deploy scripts.
**- How to verify it**
https://docs.google.com/document/d/1apFjxSa7DJBwt6L0U0TGHC56aRnRtqOqmOXgSLa2DC8/edit#

```
Debian GNU/Linux 9 sonic ttyS0

sonic login: admin
Password: 
Linux sonic 4.9.0-9-2-amd64 #1 SMP Debian 4.9.168-1+deb9u3 (2015-12-19) x86_64
You are on
  ____   ___  _   _ _  ____
 / ___| / _ \| \ | (_)/ ___|
 \___ \| | | |  \| | | |
  ___) | |_| | |\  | | |___
 |____/ \___/|_| \_|_|\____|

-- Software for Open Networking in the Cloud --

Unauthorized access and/or use are prohibited.
All access and/or use are subject to monitoring.

Help:    http://azure.github.io/SONiC/

admin@sonic:~$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
admin@sonic:~$ 

```

jleveque committed:
[sonic-slave] Update linux-compiler-gcc package version to fix build (#3514)

lguohan committed:
[build]: install cmake 3.13.2 (#4319)

fix #4315

download and install cmake 3.13.2 from azure storage 

latest cmake 3.16.3 from stretch-backports breaks the libyang 1.0.73 build

Signed-off-by: Guohan Lu <lguohan@gmail.com>

YuriiHordynskyi committed:
[T2624]: Fixed SONiC compilation for Barefoot SDE-9.0.0.
         Added IMT related customization for SONiC image deployment.

Signed-off-by: Yurii Hordynskyi <yuriih@interfacemasters.com>

Signed-off-by: Yurii Hordynskyi <yuriih@interfacemasters.com>

YuriiHordynskyi commented on platform/barefoot/bfn-platform.mk:
```diff
@@ -1,5 +1,5 @@
-BFN_PLATFORM = bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb
-$(BFN_PLATFORM)_URL = "https://github.com/barefootnetworks/sonic-release-pkgs/raw/rel_9_0/bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb"
+BFN_PLATFORM = bfnplatform_1.0.0_amd64.deb
+$(BFN_PLATFORM)_URL = "http://192.168.157.83/yuriih/barefoot/bfnplatform_1.0.0_amd64.deb"
```
Need to be changed to store the Barefoot related packages in some commonplace accessible for us and customers.

AndriiKit commented on platform/barefoot/bfn-platform.mk:
```diff
@@ -1,5 +1,5 @@
-BFN_PLATFORM = bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb
-$(BFN_PLATFORM)_URL = "https://github.com/barefootnetworks/sonic-release-pkgs/raw/rel_9_0/bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb"
+BFN_PLATFORM = bfnplatform_1.0.0_amd64.deb
+$(BFN_PLATFORM)_URL = "http://192.168.157.83/yuriih/barefoot/bfnplatform_1.0.0_amd64.deb"
```
Maybe it is better to use some general location and don't store package under your personal directory on 192.168.157.83 server

YuriiHordynskyi replied:
Please see the comment above. I said the same =)

lguohan committed:
[sonic-slave-stretch]: install same version for docker-ce and docker-ce-cli

difference versions can cause compatibility issue between the server and client

Signed-off-by: Guohan Lu <lguohan@gmail.com>

vitalivanov commented on build_debian.sh:
```diff
@@ -539,8 +543,8 @@ sudo mkdir -p $FILESYSTEM_ROOT/var/lib/docker
 
 ## Compress docker files
-pushd $FILESYSTEM_ROOT && sudo tar czf $OLDPWD/$FILESYSTEM_DOCKERFS -C ${DOCKERFS_PATH}var/lib/docker .; popd
+pushd $FILESYSTEM_ROOT && sudo env GZIP=-9 tar czf $OLDPWD/$FILESYSTEM_DOCKERFS -C ${DOCKERFS_PATH}var/lib/docker .; popd
```
@YuriiHordynskyi why do we need that?

YuriiHordynskyi replied:
To decrease image size more than 15MB.

vitalivanov commented on dockers/docker-base-stretch/Dockerfile.j2:
```diff
@@ -74,10 +74,18 @@ RUN apt-get -y purge   \
-    apt-get autoclean -y                 && \
-    apt-get autoremove -y                && \
-    rm -rf /var/lib/apt/lists/* /tmp/*
+RUN apt-get clean -y         && \
```
@YuriiHordynskyi you should submit it back to sonic as PR.

YuriiHordynskyi replied:
ok. will create the PR if it's not present on the master branch.

vitalivanov commented on platform/barefoot/bfn-platform.mk:
```diff
@@ -1,5 +1,5 @@
-BFN_PLATFORM = bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb
-$(BFN_PLATFORM)_URL = "https://github.com/barefootnetworks/sonic-release-pkgs/raw/rel_9_0/bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb"
+BFN_PLATFORM = bfnplatform_1.0.0_amd64.deb
```
we will correct this in next commits when server is ready.

YuriiHordynskyi replied:
ok

vitalivanov commented on rules/config:
```diff
@@ -36,7 +36,7 @@ SONIC_CONFIG_MAKE_JOBS = $(shell nproc)
 
 # DEFAULT_PASSWORD - default password for installer build
-DEFAULT_PASSWORD = YourPaSsWoRd
+DEFAULT_PASSWORD = admin123
```
Let's have sonic password as default. It's sonic at the end...

YuriiHordynskyi replied:
ok will revert

vitalivanov commented on rules/config:
```diff
@@ -77,7 +77,7 @@ ENABLE_ORGANIZATION_EXTENSIONS = y
 
 # ENABLE_SYSTEM_TELEMETRY - build docker-sonic-telemetry for system telemetry support
-ENABLE_SYSTEM_TELEMETRY = y
+ENABLE_SYSTEM_TELEMETRY = n
```
Why? because of size?

YuriiHordynskyi replied:
barefoot 9.0.0 did not support telemetry and compilation was broken.
